### PR TITLE
Parser for JEOL TIFF, fixes #39

### LINKED ISSUE #39: Add parsing for JEOL

### PR DESCRIPTION
**Pull request by mkuehbach**

### CODE REVIEW

File: src/pynxtools_em/parsers/image_tiff_jeol.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                and self.tmp["flat_dict_meta"]["CM_LABEL"] == "JEOL"
+            ):
+                self.supported = True
+            else:
+                self.supported = False
+                print(
+                    f"Parser {self.__class__.__name__} finds no content in {self.file_path} that it supports"
+                )
+
+    def parse(self, template: dict) -> dict:
```
Comment (lukaspie) [inline src/pynxtools_em/parsers/image_tiff_jeol.py:141]: docstring missing

File: src/pynxtools_em/parsers/image_tiff_jeol.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                        "strength": 1,
+                    }
+                    template[f"{trg}/AXISNAME[axis_{dim}]/@long_name"] = (
+                        f"Coordinate along {dim}-axis ({scan_unit[dim]})"
+                    )
+                    template[f"{trg}/AXISNAME[axis_{dim}]/@units"] = f"{scan_unit[dim]}"
+                image_identifier += 1
+        return template
+
+    def add_various_dynamic(self, template: dict) -> dict:
```
Comment (lukaspie) [inline src/pynxtools_em/parsers/image_tiff_jeol.py:229]: docstring missing

File: src/pynxtools_em/parsers/image_tiff_jeol.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        identifier = [self.entry_id, self.event_id, 1]
+        add_specific_metadata_pint(
+            JEOL_VARIOUS_DYNAMIC_TO_NX_EM,
+            self.tmp["flat_dict_meta"],
+            identifier,
+            template,
+        )
+        return template
+
+    def add_various_static(self, template: dict) -> dict:
```
Comment (lukaspie) [inline src/pynxtools_em/parsers/image_tiff_jeol.py:239]: docstring missing

**Review by lukaspie — CHANGES_REQUESTED** [review]
Mostly missing docstrings. Consider not using `print()`, but `logger.info()`

File: src/pynxtools_em/parsers/image_tiff_jeol.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                    f"Parser {self.__class__.__name__} finds no content in {self.file_path} that it supports"
+                )
+
+    def parse(self, template: dict) -> dict:
+        if self.supported is True:
+            print(f"Parsing via JEOL...")
+            # metadata have at this point already been collected into an fd.FlatDict
+            self.process_event_data_em_metadata(template)
+            self.process_event_data_em_data(template)
+        else:
```
Comment (lukaspie) [inline src/pynxtools_em/parsers/image_tiff_jeol.py:147]: logger.warning?

File: src/pynxtools_em/parsers/image_tiff_jeol.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    def check_if_tiff_jeol(self):
+        """Check if resource behind self.file_path is a TaggedImageFormat file.
+
+        This loads the metadata with the txt_file_path first to the formatting of that
+        information can be used to tell JEOL data apart from other data.
+        """
+        # currently not voting-based algorithm required as used in other parsers
+        if self.txt_file_path is None:
+            self.supported = False
+            print(
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply src/pynxtools_em/parsers/image_tiff_jeol.py:78]: #62

File: src/pynxtools_em/parsers/image_tiff_jeol.py (hunk 2 above)
Comment (mkuehbach) [reply src/pynxtools_em/parsers/image_tiff_jeol.py:229]: Fixed

File: src/pynxtools_em/parsers/image_tiff_jeol.py (hunk 4 above)
Comment (mkuehbach) [reply src/pynxtools_em/parsers/image_tiff_jeol.py:147]: #62

File: src/pynxtools_em/parsers/image_tiff_jeol.py (hunk 3 above)
Comment (mkuehbach) [reply src/pynxtools_em/parsers/image_tiff_jeol.py:239]: Fixed

File: src/pynxtools_em/parsers/image_tiff_jeol.py (hunk 1 above)
Comment (mkuehbach) [reply src/pynxtools_em/parsers/image_tiff_jeol.py:141]: Added

**Pull request merged**